### PR TITLE
feat(cysync): evm transaction support

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
implements https://app.clickup.com/t/9002019994/PRF-6285

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6285 Add support to CySync app](https://app.clickup.com/t/9002019994/PRF-6285)

File: packages/cysync-core/src/constants/i18n/en.ts [hunk 1]
```diff
@@ -469,6 +469,10 @@ const en = {
         { id: 2, name: 'Average' },
         { id: 3, name: 'Max' },
       ],
+      inputLabels: {
+        priceLabel: 'Gas Price',
+        limitLabel: 'Gas Limit',
+      },
```
Comment (irshadCypherock) [inline packages/cysync-core/src/constants/i18n/en.ts]:
```suggestion
      inputLabels: {
        gasPrice: 'Gas Price',
        gasLimit: 'Gas Limit',
      },
```

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx [hunk 2]
```diff
@@ -34,12 +37,31 @@ export const SingleTransaction: React.FC = () => {
     prepare(txn);
   }, []);
 
+  const bitcoinMaxSendAmount = (txn: IPreparedTransaction) => {
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx]:
```suggestion
  const getBitcoinMaxSendAmount = (txn: IPreparedTransaction) => {
```

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx [hunk 3]
```diff
@@ -34,12 +37,31 @@ export const SingleTransaction: React.FC = () => {
...
+    const { computedData } = txn as IPreparedBtcTransaction;
+    return computedData.outputs[0]?.value.toString() || '';
+  };
+
+  const evmMaxSendAmount = (txn: IPreparedTransaction) => {
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx]:
```suggestion
  const getEvmMaxSendAmount = (txn: IPreparedTransaction) => {
```

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx [hunk 4]
```diff
@@ -34,12 +37,31 @@ export const SingleTransaction: React.FC = () => {
...
+    const { computedData } = txn as IPreparedEvmTransaction;
+    return computedData.output.amount;
+  };
+
+  const computedAmounMap: Record<
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx]:
```suggestion
  const computedAmountMap: Record<
```

File: packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx [hunk 5]
```diff
@@ -66,6 +69,29 @@ export const SummaryDialog: React.FC = () => {
     return details ?? [];
   };
 
+  const bitcoinFeeAmount = (txn: IPreparedTransaction | undefined) => {
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx]:
```suggestion
  const getBitcoinFeeAmount = (txn: IPreparedTransaction | undefined) => {
```

File: packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx [hunk 6]
```diff
@@ -66,6 +69,29 @@ export const SummaryDialog: React.FC = () => {
...
+    // return '0' in error scenarios because BigNumber cannot handle empty string
+    if (!txn) return '0';
+    const { computedData } = txn as IPreparedBtcTransaction;
+    return computedData.fee.toString() || '0';
+  };
+
+  const evmFeeAmount = (txn: IPreparedTransaction | undefined) => {
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx]:
```suggestion
  const getEvmFeeAmount = (txn: IPreparedTransaction | undefined) => {
```

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/FeeSection/EthereumInput.tsx [hunk 7]
```diff
@@ -1,58 +1,63 @@
...
-    onGasLimitChange(numberValue);
+    // on change gas limit, prepare EVM fee
+    onChange({ gasLimit: numberValue });
   };
 
+  useEffect(() => {
+    if (transaction.userInputs.gasPrice) {
+      // user input gas price should reflect in the ui
+    }
+  });
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/Components/FeeSection/EthereumInput.tsx]: Remove this

File: packages/coin-support-evm/src/operations/prepareTransaction/index.ts [hunk 8]
```diff
@@ -46,38 +46,41 @@ export const prepareTransaction = async (
...
+    .minus(output.amount || '0')
     .minus(fee)
     .isPositive();
 
+  if (txn.userInputs.isSendAll && hasEnoughBalance) {
+    output.amount = new BigNumber(account.balance).minus(fee).toString(10);
+    // update userInput so that the max amount is editable & not reset to 0
+    txn.userInputs.outputs[0].amount = output.amount;
+  }
+  txn.userInputs.gasLimit = gasLimit;
```
Comment (irshadCypherock) [inline packages/coin-support-evm/src/operations/prepareTransaction/index.ts]: Logic looks incorrect for sendAll when amount is specificed

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/index.tsx [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
 
+import { useSendDialog } from '~/dialogs/Send/context';
 import { selectLanguage, useAppSelector } from '~/store';
 
-import { BatchTransactionBody } from './BatchTransactionBody';
+import { BatchTransaction } from './BatchTransaction';
 import { SingleTransaction } from './SingleTransaction';
 
-export const AddressAndAmountSection: React.FC = () => {
+const BitcoinAnA: React.FC = () => {
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/index.tsx]: Update this for other variables as well
```suggestion
const BitcoinAddressnAmount: React.FC = () => {
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/coin-support-evm/src/operations/prepareTransaction/index.ts (hunk 8 above)
Comment (ujjwal-cyph) [reply packages/coin-support-evm/src/operations/prepareTransaction/index.ts]: good catch. thanks. updated

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/FeeSection/EthereumInput.tsx (hunk 7 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/Components/FeeSection/EthereumInput.tsx]: updated

File: packages/cysync-core/src/constants/i18n/en.ts (hunk 1 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/constants/i18n/en.ts]: changed

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx (hunk 2 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx]: changed

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx (hunk 3 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx]: renamed

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx (hunk 4 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/SingleTransaction.tsx]: corrected

File: packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx (hunk 5 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx]: renamed

File: packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx (hunk 6 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/SummaryDialog.tsx]: renamed

File: packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/index.tsx (hunk 9 above)
Comment (ujjwal-cyph) [reply packages/cysync-core/src/dialogs/Send/Dialogs/Components/AddressAndAmountSection/index.tsx]: renamed

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**